src/main/java/duke/logic/Logic.java

```diff
@@ -0,0 +1,77 @@
+package duke.logic;
+
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.command.CommandProcessor;
+import duke.exceptions.HalException;
+import duke.storage.Storage;
+import javafx.application.Platform;
+
+/**
+ * The `UserInterface` class handles user interactions and serves as the main interface for the Duke application.
+ */
+public class Logic {
+
+    private Scanner scan;
+    private CommandProcessor cmd;
+    private boolean startUpSuccess = false;
```

Comment:
Would it be better to use a more boolean variable name like 'isStartUpSuccess'?

Reply:
Yep! Thanks for catching it